src/aiori-IME.c

```diff
@@ -100,92 +105,113 @@ ior_aiori_t ime_aiori = {
         .create        = IME_Create,
         .open          = IME_Open,
         .xfer          = IME_Xfer,
+        .xfer_hints    = IME_Xferhints,
         .close         = IME_Close,
         .delete        = IME_Delete,
         .get_version   = IME_GetVersion,
         .fsync         = IME_Fsync,
         .get_file_size = IME_GetFileSize,
         .access        = IME_Access,
-        .statfs        = IME_StatFS,
-        .rmdir         = IME_RmDir,
-        .mkdir         = IME_MkDir,
+        .statfs        = IME_Statfs,
+        .rmdir         = IME_Rmdir,
+        .mkdir         = IME_Mkdir,
         .stat          = IME_Stat,
         .initialize    = IME_Initialize,
         .finalize      = IME_Finalize,
-        .get_options   = IME_options,
+        .get_options   = IME_Options,
 #if (IME_NATIVE_API_VERSION >= 132)
         .sync          = IME_Sync,
         .mknod         = IME_Mknod,
 #endif
         .enable_mdtest = true,
 };
 
+static aiori_xfer_hint_t *hints = NULL;
+static bool ime_initialized = false;
+
+
 /***************************** F U N C T I O N S ******************************/
 
+void IME_Xferhints(aiori_xfer_hint_t *params)
+{
+        hints = params;
+}
+
 /*
  * Initialize IME (before MPI is started).
  */
-static void IME_Initialize()
+void IME_Initialize()
 {
+        if (ime_initialized)
+                return;
+
         ime_native_init();
+        ime_initialized = true;
 }
 
 /*
  * Finlize IME (after MPI is shutdown).
  */
-static void IME_Finalize()
+void IME_Finalize()
 {
+        if (!ime_initialized)
+                return;
+
         (void)ime_native_finalize();
+        ime_initialized = true;
 }
 
 /*
  * Try to access a file through the IME interface.
  */
-static int IME_Access(const char *path, int mode, IOR_param_t *param)
+
+int IME_Access(const char *path, int mode, aiori_mod_opt_t *module_options)
 {
-        (void)param;
+        (void)module_options;
```

Comment:
Note: this is for silencing compiler warnings/errors.

Reply:
Yes, I'll add a comment here.